pkg/executor/analyze.go

```diff
@@ -173,7 +173,8 @@ TASKLOOP:
 	if err != nil {
 		sessionVars.StmtCtx.AppendWarning(err)
 	}
-	return statsHandle.Update(ctx, infoSchema)
+	statsHandle.TriggerSyncStats()
```

Comment:
Here we need to make sure that it definitely executes successfully, otherwise we have the problem of a table being analyzed but immediately analyzed again. This is because our auto-analysis ticker will be triggered again during the analysis, and then because it didn't update the stats cache correctly to load the count and modify_count of the analyzed table, it will use the last count and modify_count to calculate the change ratio, which will ultimately cause it to queue up again.

Reply:
add force sync stats before the auto-analyze.

Reply:
Generally, the load time for this is in the millisecond range, and the gap time can be considered negligible.